File: ZANT/frontend/src/app/models/accident-report.ts
import { PersonData } from './person-data';
import { BusinessData } from './business-data';
import { AccidentData } from './accident-data';
import { Witness } from './witness';

export interface AccidentReport {
  id?: string;
  personData?: PersonData;
  businessData?: BusinessData;
  accidentData?: AccidentData;
  witnesses?: Witness[];
  proxyData?: PersonData;
  isFiledByProxy?: boolean;
  requiredDocuments?: string[];
}

File: ZANT/backend/src/main/java/com/zant/backend/model/AccidentReport.java
package com.zant.backend.model;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
public class AccidentReport {

    private PersonData victimData;
    private BusinessData businessData;
    private AccidentData accidentData;
    private List<Witness> witnesses;
    private PersonData attorneyData;
    private List<String> requiredDocuments;
}
